docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_log_file_integrity_remediation.mdx

```diff
@@ -1,6 +1,215 @@
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and navigate to the CloudTrail service. 
+
+2. In the CloudTrail dashboard, click on "Trails" in the left navigation pane. This will display a list of all the trails in your AWS account.
+
+3. Click on the name of the trail you want to check. This will open the trail's details page.
+
+4. In the trail's details page, look for the "File integrity validation" section. If the status is "Yes", then the file integrity validation feature is enabled for the trail. If the status is "No", then the feature is not enabled.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine. You can download it from the official AWS website. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your AWS Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. List all trails: Use the following AWS CLI command to list all the trails in your AWS account:
+
+   ```
+   aws cloudtrail describe-trails
+   ```
+   This command will return a JSON output with details of all the trails.
+
+3. Check File Integrity Validation Feature: For each trail, check if the `LogFileValidationEnabled` attribute is set to `true`. This attribute indicates whether the file integrity validation feature is enabled for the trail. You can do this by running the following command:
+
+   ```
+   aws cloudtrail get-trail-status --name TrailName
+   ```
+   Replace `TrailName` with the name of your trail. If the `LogFileValidationEnabled` attribute is set to `true`, the file integrity validation feature is enabled for the trail.
+
+4. Repeat the process: Repeat step 3 for each trail in your AWS account. If any trail has the `LogFileValidationEnabled` attribute set to `false`, it means the file integrity validation feature is not enabled for that trail, indicating a misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: Before you start, you need to install the AWS SDK for Python (Boto3) in your environment. This can be done using pip:
+
+   ```bash
+   pip install boto3
+   ```
+
+2. Import the necessary libraries and set up AWS credentials: In your Python script, you need to import the Boto3 library and set up your AWS credentials (Access Key ID and Secret Access Key). You can do this using the AWS CLI or by manually setting them in your script:
+
+   ```python
+   import boto3
+
+   # Set AWS credentials
+   aws_access_key_id = 'YOUR_ACCESS_KEY'
+   aws_secret_access_key = 'YOUR_SECRET_KEY'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>